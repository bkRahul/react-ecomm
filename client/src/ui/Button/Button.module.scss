.Button {
	min-width: 165px;
	width: auto;
	height: 50px;
	letter-spacing: 0.5px;
	line-height: 50px;
	padding: 0 35px 0 35px;
	font-size: 15px;
	background-color: black;
	color: white;
	text-transform: uppercase;
	font-family: 'Open Sans Condensed';
	font-weight: bolder;
	border: none;
	cursor: pointer;

	&:hover {
		background-color: white;
		color: black;
		border: 1px solid black;
	}

	&.GoogleBtn {
		background-color: #4285f4;
		border: none;
	}
	&.ClearCartBtn {
		min-width: unset;
		background-color: #fff;
		color: #000;
		padding: 0 15px;
		border: 1px solid black;
	}
	&.CheckoutCartBtn {
		min-width: unset;
		padding: 0 15px;
	}
	&.ProductBtn {
		color: black;
		background-color: #fff;
		opacity: 0.8;
		position: absolute;
		top: 250px;
		display: none;
		border: 1px solid #000;

		&:hover {
			background-color: #000;
			color: #fff;
		}
	}
}
