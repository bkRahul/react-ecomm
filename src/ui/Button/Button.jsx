import React from 'react';
import classes from './Button.module.scss';

export const Button = ({ disabled, btnType, clicked, children }) => {
  return (
    <button
      disabled={disabled}
      className={[classes.Button, classes[btnType]].join(' ')}
      onClick={clicked}
    >
      {children}
    </button>
  );
};
