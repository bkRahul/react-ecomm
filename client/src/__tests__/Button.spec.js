import React from 'react'
import { shallow } from 'enzyme'
import { Button } from '../ui/Button/Button'

describe('Button component', () => {
	it('to render correctly', () => {
		expect(shallow(<Button />).length).toEqual(1)
	})

	it('to match snapshot', () => {
		expect(shallow(<Button />)).toMatchSnapshot()
	})
})
