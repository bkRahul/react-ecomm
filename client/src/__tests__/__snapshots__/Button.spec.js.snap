// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component to match snapshot 1`] = `ShallowWrapper {}`;
